src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelRunEventsHandler.cs

```diff
@@ -174,6 +175,17 @@ public int LaunchProcessWithDebuggerAttached(TestProcessStartInfo testProcessSta
             return this.actualRunEventsHandler.LaunchProcessWithDebuggerAttached(testProcessStartInfo);
         }
 
+        /// <inheritdoc />
+        public bool AttachDebuggerToProcess(int pid)
+        {
+            if (!(this.actualRunEventsHandler is ITestRunEventsHandler2))
```

Comment:
is as check here as well and exception wording

Reply:
Done.